### PR TITLE
Postgres example

### PR DESCRIPTION
**Pull request by thor314**
I'm working on adding an example for postgres, since it seems to differ a bit from the sqlite example provided. I'm currently debugging why the session authentication here:
```rust
// todo: currently returning Err
        let user = match auth_session.authenticate(creds.clone()).await {
            Ok(Some(user)) => user,
            Ok(None) => {
                messages.error("Invalid credentials");

                let mut login_url = "/login".to_string();
                if let Some(next) = creds.next {
                    login_url = format!("{}?next={}", login_url, next);
                };

                return Redirect::to(&login_url).into_response();
            }
            Err(_) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
        };
```

is failing. Marking as draft PR.

### CODE REVIEW

**Comment by thor314** [comment]
I wrote this postgres example to document a bug I'm running into in my own [repo](https://github.com/thor314/zkhn-rust-api). This PR creates an example for `postgres`, and documents a failing assumption--that a table called `session` exists in the database.

The following login method fails accordingly:
```rust
    pub async fn login(
        mut auth_session: AuthSession,
        messages: Messages,
        Form(creds): Form<Credentials>,
    ) -> impl IntoResponse {
        // todo: see auth session:
        // auth session thinks a table called tower_sessions exists:
        // schema_name: "tower_sessions",
        // table_name: "session",
        #[cfg(test)]
        dbg!(auth_session.clone());

        let user = match auth_session.authenticate(creds.clone()).await {
```

File: examples/postgres/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#[cfg(test)]
+mod test;
+mod users;
+mod web;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::registry()
+        .with(EnvFilter::new(std::env::var("RUST_LOG").unwrap_or_else(
+            |_| "axum_login=debug,tower_sessions=debug,sqlx=warn,tower_http=debug".into(),
```
Comment (maxcountryman) [inline examples/postgres/src/main.rs:19]: I would recommend increasing the sqlx log level while you're tracking down the issue.

**Comment by thor314** [comment]
with the last 3 commits, I've updated the postgres example to pass unit tests.

**Comment by maxcountryman** [comment]
This should probably be an independent repository, perhaps even a fully-fledged template others could build from. I don't think it's prudent to increase the maintenance burden of `axum-login` itself so I'm going to close this. If you end up establishing a template repo, please feel free to link to it from the `axum-login` README.

**Pull request closed without merging**